src/storage/src/render/upsert.rs

```diff
@@ -333,11 +419,16 @@ where
                     state
                         .multi_get(
                             multi_get_scratch.drain(..),
-                            commands_state.iter_mut().map(|(_, v)| v),
+                            commands_state.values_mut().map(|v| &mut v.value),
                         )
                         .await
                         .expect("hashmap impl to not fail");
 
+                    // Update state with calculated sizes in bytes
+                    commands_state
+                        .values_mut()
+                        .for_each(|v| v.populate_initial_size());
```

Comment:
can we do this per-value in the `commands.next()` loop below?

Reply:
Oh yeah, that would definitely be better. Let me change this.

Reply:
We are doing a dedup before the `commands.next`, so I guess we'll never have repeated values for the same key, so doing populate should be fine

Reply:
Oh but we might still have some null values, so would not work

Reply:
(chatted in slack)